File: src/Web/WebApi/src/service/PartyPurposeService.js
export default class PartyPurposeService {
    constructor(request) {
        this.request = request;
    }

    async savePartyPurposes(email, purposeTypes) {
        const response = await this.request({
            method: 'post',
            url: '/partyPurpose/save',
            data: { email, purposeTypes }
        });
        return response?.data ?? response;
    }
}

File: src/Mobile/lib/services/party_purpose_service.dart
import 'package:dio/dio.dart';

class PartyPurposeService {
  PartyPurposeService(this._dio);

  final Dio _dio;

  Future<dynamic> savePartyPurposes({
    required String email,
    required List<int> purposeTypes,
  }) async {
    final r = await _dio.post<dynamic>(
      '/partyPurpose/save',
      data: {'email': email, 'purposeTypes': purposeTypes},
    );
    return r.data;
  }

  Future<dynamic> getMyPurposes({String? email}) async {
    final r = await _dio.get<dynamic>(
      '/partyPurpose/my-purposes',
      queryParameters: {
        if (email != null && email.trim().isNotEmpty) 'email': email.trim(),
      },
      options: Options(
        validateStatus: (s) => s != null && s >= 200 && s < 300,
      ),
    );
    return r.data;
  }
}
